### PR TITLE
kuberay compatibility test with ray

### PR DESCRIPTION
**Pull request by wilsonwang371**
## Why are these changes needed?

Kuberay compatibility test with Ray

## Related issue number

#146 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(
   - [X] Partially manual tested

### CODE REVIEW

File: tests/compatibility-test.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+
+def create_cluster():
+    shell_assert_success(
+        'kind create cluster --config {}'.format(kindcluster_config_file))
+
+
+def apply_kuberay_resources():
+    shell_assert_success(
+        'kubectl apply -k github.com/ray-project/kuberay/manifests/cluster-scope-resources')
```
Comment (Jeffwan) [inline tests/compatibility-test.py:43]: Did we update the image `kubectl kustomize edit set image kuberay/operator=kuberay/operator:${commit_hash}` here to test against nightly image?

File: tests/compatibility-test.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+def delete_cluster():
+    shell_assert_success('kind delete cluster')
+
+
+def download_images():
+    client = docker.from_env()
+    client.images.pull(ray_image)
+    # not enabled for now
+    # shell_assert_success('kind load docker-image \"{}\"'.format(ray_image))
```
Comment (Jeffwan) [inline tests/compatibility-test.py:79]: should we load the image CI just build into the kind env?

File: tests/compatibility-test.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+import docker
+
+ray_version = '1.8.0'
+ray_image = "rayproject/ray:1.8.0"
+
+kindcluster_config_file = 'tests/config/cluster-config.yaml'
+
+raycluster_spec_template = 'tests/config/ray-cluster.mini.yaml.template'
+raycluster_service_file = 'tests/config/raycluster-service.yaml'
```
Comment (Jeffwan) [inline tests/compatibility-test.py:18]: I can not find the usage of this file. What's the usage here?

File: .github/workflows/test-job.yaml [hunk 4]
```diff
@@ -109,6 +111,31 @@ jobs:
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Python dependencies
+      run: pip install -r ./tests/config/requirements.txt
+
+    - name: Run compatibility test
+      run: |
+        for i in "rayproject/ray:1.8.0" "rayproject/ray:1.9.0" "rayproject/ray:1.10.0"
```
Comment (Jeffwan) [inline .github/workflows/test-job.yaml:133]: let's use 1.8.0 1.9.0 to emphasize versions, KUBERAY_TEST_RAY_IMAGE=rayproject/ray:${version}?

**Review by Jeffwan — COMMENTED** [review]
excellent work, Wilson! I leave some comments and please take a look

File: tests/config/ray-cluster.mini.yaml.template [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    serviceType: ClusterIP
+    # the pod replicas in this group typed head (assuming there could be more than 1 in the future)
+    replicas: 1
+    # logical group name, for this called head-group, also can be functional
+    # pod type head or worker
+    # rayNodeType: head # Not needed since it is under the headgroup
+    # the following params are used to complete the ray start: ray start --head --block --redis-port=6379 ...
+    rayStartParams:
+      port: '6379' # should match headService targetPort
+      object-manager-port: '12345'
```
Comment (Jeffwan) [inline tests/config/ray-cluster.mini.yaml.template:23]: `object-manager-port`, `node-manager-port` and `object-store-memory` brings the problem in 1.9.x.

Comment (wilsonwang371) [reply tests/config/ray-cluster.mini.yaml.template]: just saw this....

File: tests/compatibility-test.py (hunk 3 above)
Comment (wilsonwang371) [reply tests/compatibility-test.py:18]: raycluster-service.yaml was the same as the one from patch #147

File: tests/compatibility-test.py (hunk 2 above)
Comment (wilsonwang371) [reply tests/compatibility-test.py:79]: > should we load the image CI just build into the kind env?

kuberay will automatically download this image, right?

Comment (Jeffwan) [reply tests/compatibility-test.py:79]: yes. it downloads ray:1.9.0 etc but we want to test against new built operator image which should be loaded into kind cluster

**Review by Jeffwan — APPROVED** [review]

**Review by Jeffwan — APPROVED** [review]

**Review by Jeffwan — APPROVED** [review]
Great work!

**Pull request merged**